src/lib.rs

```diff
@@ -1,14 +1,248 @@
+        self.pwm.use_external_clock().unwrap();
+        self.pwm.set_prescale(100).unwrap();
+        self.pwm.enable().unwrap();
+
+        self.bmp.zero().unwrap();
+
+        self.led.all_off()
+    }
+
+    pub fn self_test(&mut self) -> bool {
+        //Check if the sensors are attached by it's IDs,
+        //run self-test if they have.
+        self.imu.check_identity(&mut Delay {}).unwrap();
+        self.mag
+            .self_test()
+            .expect("Error : Error on magnetometer during self-test")
+    }
+
+    pub fn set_pwm_value(&mut self) {
+        let channel = pwm_Channel::C0; //todo import enums
+        let value = 2047; //todo a fomula that validades from 0 to 100.00% and convert to 4092
+        self.pwm.set_channel_on(channel, 0).unwrap(); //Channel::C0
+        self.pwm.set_channel_off(channel, value).unwrap();
+    }
+
+    pub fn set_pwm_freq(&mut self) {
+        todo!()
+    }
+
+    pub fn set_pwm_off(&mut self) {
+        todo!()
+    }
+
+    pub fn set_pwm_on(&mut self) {
+        todo!()
+    }
+
+    pub fn set_led_on(&mut self) {
+        self.led.all_on()
+    }
+    pub fn set_led_off(&mut self) {
+        self.led.all_off()
+    }
+
+    pub fn read_mag(&mut self) -> [f32; 3] {
+        let (x, y, z) = self.mag.read().unwrap();
+        [x.into(), y.into(), z.into()]
+    }
+
+    pub fn read_bmp(&mut self) -> [f32; 3] {
+        let pressure = self.bmp.pressure_kpa().unwrap();
+        let altitude = self.bmp.altitude_m().unwrap();
+        let temperature = self.bmp.temperature_celsius().unwrap();
+        [altitude, pressure, temperature] //fix here struc.
+    }
+
+    pub fn read_temperature(&mut self) -> f32 {
+        //this come from bmp device
+        self.bmp.temperature_celsius().unwrap()
+    }
+
+    pub fn read_altitude(&mut self) -> f32 {
+        //this come from bmp device
+        self.bmp.altitude_m().unwrap()
+    }
+    pub fn read_pressure(&mut self) -> f32 {
+        //this come from bmp device
+        self.bmp.pressure_kpa().unwrap()
+    }
+
+    pub fn read_adc(&mut self) -> [i16; 4] {
+        let values: [i16; 4] = [
+            block!(self.adc.read(&mut channel::SingleA0)).unwrap(),
+            block!(self.adc.read(&mut channel::SingleA1)).unwrap(),
+            block!(self.adc.read(&mut channel::SingleA2)).unwrap(),
+            block!(self.adc.read(&mut channel::SingleA3)).unwrap(),
+        ];
+        values
+    }
+    pub fn read_accel(&mut self) -> [f32; 3] {
+        self.imu.get_scaled_accel().unwrap()
+    }
+    pub fn read_gyro(&mut self) -> [f32; 3] {
+        self.imu.get_scaled_gyro().unwrap()
+    }
+    pub fn read_all(&mut self) -> SensorData {
+        SensorData {
+            adc: self.read_adc(),
+            temperature: self.read_temperature(),
+            pressure: self.read_pressure(),
+            accelerometer: self.read_accel(),
+            magnetometer: self.read_mag(),
+            gyro: self.read_gyro(),
+        }
     }
 }
```

Comment:
It isn't too important at this stage, but it's good to think that in the future we might prefer to return errors to the caller instead of panicking at every single possible result.